commands/setUser.js

```diff
@@ -349,4 +349,124 @@ const updateUserNickname = async function(userid, nickname, server) {
+   * 1) The assigner has the Committee President or PVP role
+   * 2) If the assigner is a Committee President, they have exactly one <Committee> Officer role
+   * 3) If an assignee does not have the ACM Officer role, then they also do not have a
+   *    <Committee> Officer role
+   *
+   * @param assigner - the GuildMember object representing the person calling the command
+   * @param assigneeInfo - username or userid of user whose role is to be assigned
+   * @param server - current Discord.Guild object
+   * @returns {Array} - contents: [error object, message]
+   */
+
+  // If userid not supplied, query from database
+  let assigneeId;
+  if (assigneeInfo.match('.+#([0-9]){4}')) {
+    const [username, discriminator] = assigneeInfo.split('#');
+    const db = await sqlite.open({
+      filename: config.db_path,
+      driver: sqlite3.Database,
+    });
+    let row;
+    try {
+      row = await db.get(
+        `
+        SELECT userid
+        FROM users
+        WHERE
+            username = ? AND
+            discriminator = ?
+        `,
+        [username, discriminator]
+      );
+      if (!row) return [null, 'Invalid/unverified user.'];
+      assigneeId = row.userid;
+    } catch (e) {
+      console.error(e.toString());
+      return [{ message: e.toString() }, null];
+    } finally {
+      await db.close();
+    }
+  } else {
+    if (!assigneeInfo.match('([0-9])+')) {
+      return [
+        null,
+        'Invalid user format. Please supply either a user in <username>#<discriminator> format or a user ID'
+      ];
+    }
+    assigneeId = assigneeInfo;
+  }
+
+  // Get assignee GuildMember object
+  let assignee;
+  try {
+    assignee = await server.members.fetch(assigneeId);
+  } catch(e) {
+    console.error(e);
+    return [{ message: e.toString() }, 'Invalid/unverified user.'];
+  }
+
+  // Determine assigner's committee
+  let assignerCommittee;
+  if (assigner.roles.cache.some(role => role.name === 'PVP')) {
+    // PVP assigns Board officers
+    assignerCommittee = 'Board';
+  } else {
+    for (const committee of config.committees_list) {
+      if (assigner.roles.cache.some(role => role.name === `${committee} Officer`)) {
+        assignerCommittee = committee;
+        break;
+      }
+    }
+    if (!assignerCommittee) {
+      return [
+        null,
+        'Action not permitted. If you are a committee president, make sure that you also have ' +
+        'a committee officer role before running this command.'
+      ];
+    }
+  }
+  // Officer roles to be assigned based on assigner's committee
+  const acm_officer_role = server.roles.cache.find(role => role.name === 'ACM Officer');
+  const comm_officer_role = server.roles.cache.find(role => role.name === `${assignerCommittee} Officer`);
+
+  // Toggle officer roles
+  if (assignee.roles.cache.some(role => role.name === acm_officer_role.name)) {
+    if (assignee.roles.cache.some(role => role.name === comm_officer_role.name)) {
```

Comment:
I think some of the logic is incorrect because a user can be an officer for multiple committees. For example, if a user has the ACM Officer role and the ACM Hack Officer role, and now the ICPC president/officer now wants to add the user as an ACM ICPC Officer role, in the current code, it wouldn't be able to, when in reality they should be able to. Instead, you want to just toggle it i.e.

My suggestion for logic is
Is ACM officer 
   1. Yes -> Is already in targeted committee
      1. Yes -> toggle remove role for targeted committee and if committees = 0, remove ACM officer
      2. No -> toggle add role for targeted committee
   2. No -> Is in a committee
      1. Yes -> Is targeted committee
         1. Yes -> Toggle remove role for targeted committee and if committees = 0, remove ACM officer
         2. No -> Toggle add role for targeted committee and ACM officer
      2. No -> Toggle add role for targeted committee and ACM officer

Reply:
this makes more sense than the current logic to me, yes. I think we should change it

Reply:
Alright makes sense. I'll do that